### ISSUE TITLE
Installer needs fixing

### ISSUE THREAD

**Issue by JeDa** [WWXD/WorldWeb-XD#42]
I was gonna try out installing WWXD locally. It did land me to /preinstall.php

After I did put the MySQL, it did ask me the owner credentials - but before I could put anything I was redirected to /install.php! And if you tried again that'd be like a loop.

**Comment by maorninja** [comment]
Yeah, that's something on the PHP side of things. Most likely, its the <button> tag, we should change that with something else that won't imput the form.

**Comment by JeDa** [comment]
I did this [little workaround](http://termbin.com/5rn6) but after that the /index.php/config/database.php error then:
`Fatal error: Only variables can be passed by reference in /home/jeda/www/board/lib/yaml.php on line 732`

Yeah, the code needs some fixing.

**Comment by Repflez** [comment]
I'm pretty sure the installer doesn't need the yaml library at all.

Rewriting the installer is an option.

**Comment by JeDa** [comment]
@Repflez actually that was the board and not the installer, but yeah, installer would be good if it'd be rewritten.

**Comment by maorninja** [comment]
Yeah, we should rewrite it. But I want it to be like the ABXD one, where there are some pages for the install...

**Comment by Repflez** [comment]
@JeDa Are you running php with `E_ALL ^ E_STRICT` enabled? It shouldn't crash and burn with it.

**Comment by JeDa** [comment]
@Repflez it is a fatal error, the board doesn't loads because that error in yaml.php. using PHP 7.0 btw

**Comment by Repflez** [comment]
@JeDa Huh, Will look into it in php 7.1.15 later this week (or next.)

**Comment by maorninja** [comment]
@JeDa Does it work now?

**Comment by JeDa** [comment]
`Fatal error: Only variables can be passed by reference in /home/jeda/www/otherboard/lib/yaml.php on line 738`

nope

**Comment by maorninja** [comment]
Ahh, maybe it's because I changed `preg_replace` to `str_replace`.

Let me go fix that, brb...

**Comment by JeDa** [comment]
`Fatal error: Only variables can be passed by reference in /home/jeda/www/otherboard/lib/yaml.php on line 780`

just replace everything (?

**Comment by JeDa** [comment]
`Fatal error: require_once(): Failed opening required '/home/jeda/www/otherboard/lib/smarty/libs/Smarty.class.php' (include_path='.:/usr/share/php') in /home/jeda/www/otherboard/lib/common.php on line 135`
